orchagent/notifications.cpp

```diff
@@ -40,6 +43,11 @@ void on_switch_shutdown_request(sai_object_id_t switch_id)
     /* TODO: Later a better restart story will be told here */
     SWSS_LOG_ERROR("Syncd stopped");
 
+    if (gSwitchOrch->isFatalEventReceived())
+    {
+        abort();
```

Comment:
could you explain the flow here? not clear on the `check `and `abort `action

Reply:
Yes.
`gSwitchOrch->isFatalEventReceived()` indicates whether there has been at least one fatal event received.
`abort` is used to trigger tech support dump when auto tech support dump is enabled. The logic is:
1. The `abort` triggers `SIGABRT` signal which triggers a core dump.
2. Tech support dump registers a coredump handler and collect techsupport dump on receiving a coredump.

This is the same usage as [handleSaiFailure](https://github.com/sonic-net/sonic-swss/blob/master/orchagent/saihelper.cpp#L803)

Reply:
Does SDK push the ASIC health event notification before any shutdown event?

Reply:
Yes. This is actually guaranteed by the vendor SAI

Reply:
Hi, isn't it better to take the SAI dump before orchagent aborts? I see that is the way it is done in handlSaiFailure. Abort() is called after taking the SAI dump vs in this case where orchagent abort() can potentially interfere with SAI dump capture later. 

Also, please add a log message explicitly stating that orchagent is aborting due to a fatal ASIC/SDK health event.

Reply:
Hi @prabhataravind 
Regarding taking the SAI dump, vendor SAI should do it, which was mentioned in the HLD and discussed during the review meeting.
Usually, the events, especially the fatal ones, were triggered due to some unrecoverable situations, which means the vendor can shut down immediately after sending the fatal event. The vendor SAI should collect the dump in the same context where the fatal event is triggered. Otherwise, it requires extra communicate between swss and syncd, which takes time during which important information can be lost.
Unlike the `handlSaiFailure`, which vendor SAI doesn't ask for shutting down, and the swss has time to ask SAI for collecting dump

Regarding the log, will do.